rust/lance-index/src/scalar.rs

```diff
@@ -477,13 +486,92 @@ impl AnyQuery for LabelListQuery {
     }
 }
 
+/// A query that a NGramIndex can satisfy
+#[derive(Debug, Clone, PartialEq)]
+pub enum TextQuery {
+    /// Retrieve all row ids where the text contains the given string
+    StringContains(String),
+    // TODO: In the future we should be able to do string-insensitive contains
+    // as well as partial matches (e.g. LIKE 'foo%') and potentially even
+    // some regular expressions
```

Comment:
We should also be able to accelerate regular equality queries out of this index.

Since AFAIK we only support one index type per column, will it be annoying if users need to pick between faster "contains" and faster equality, by choosing between ngram and btree/bitmap?

Reply:
Yeah, my hope was that we could support multiple indices (at least multiple index types) on a single column but it might be useful to do equality with an ngram index since it should be much smaller for big text.

Reply:
I'm going to add support for equality and IS NULL in a future PR